pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -211,6 +211,122 @@ func TestReconcile_MissingTasks(t *testing.T) {
 	}
 }
 
+func TestReconcile_PipelineParams(t *testing.T) {
```

Comment:
i think im giving you mixed messages about when to use table driven tests and when not to, but I think this could be a candidate :thinking: 

What about this:
- Have only 1 `reconcile` test case, which DOES include a service account
- Have a smaller unit test, something that calls a function directly and doesn't have to include the entire reconcile loop, that makes sure we can handle both cases: with a service account and without a service account?

Reply:
> Have a smaller unit test, something that calls a function directly and doesn't have to include the entire reconcile loop, that makes sure we can handle both cases: with a service account and without a service account?

I am not a fan of testing private function by itself. That was reasoning behind writing a big unit test to cover that logic.

> Have only 1 reconcile test case, which DOES include a service account
Sure.